packages/storybook/src/stories/playground/customize-components.stories.tsx

```diff
@@ -465,3 +468,22 @@ CustomPanelInFooter.args = {
     Header: GridToolbar,
   },
 };
+
+export function CustomLayout() {
+  const apiRef = useApiRef();
+  apiRef.current.id = 123;
+
+  return (
+    <React.StrictMode>
+      <ApiRefProvider apiRef={apiRef}>
```

Comment:
Why not only? What's the use case for two APIs (a hook + a provider)? Also, to make it more generic:

```suggestion
      <GridProvider>
```

Reply:
Well if they want to use the apiRef and the provider within a component they have to use both.
Ie you use the provider so the underlying component can access the apiref context and state of the grid,
you use the apiRef if you need to update rows for example. So how do you have both behaviour at the same time?

Reply:
> Well if they want to use the apiRef and the provider within a component they have to use both.

Do we need to support this case? What would be wrong with the redux approach? Meaning moving the context provider to the parent so you can use the context consumer on the child.

Reply:
I also like the `<GridProvider>` more, but I think it would be even better if inside the grid provider we have the following:

```
function GridProvider({children}) {
  return (
    <GridApiContext.Provider>
      <GridStateContext.Provider>
        {children}
      </GridStateContext.Provider>
    </GridApiContext.Provider>
  );
}
```

Because then if someone wants to use the state separate from the API they can and we will still provide a nicer public API.